### PR TITLE
[SYCL] Fix get() method for non-opencl backends

### PR DESCRIPTION
**Pull request by alexanderfle**
Before this patch, if you use non-opencl backend, then get() method returns casted wrapper of the native handle and SYCL user can’t use it in his own code. As the user can't release the returned object, it used to lead to a memory leak.

Since, get() method is applicable only for OpenCL backend, this patch adds an exception for cases when get() method is called for non-opencl backend.

Signed-off-by: Alexander Flegontov <alexander.flegontov@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
looks good to me.
@smaslov-intel would you please take a look at the changes as well?
there are no any plans to support interoperability for level-zero, right?

**Comment by smaslov-intel** [comment]
> @smaslov-intel would you please take a look at the changes as well?

LGTM. These OpenCL-only .get() methods will be removed in SYCL2020 in favor of .get_native().

> there are no any plans to support interoperability for level-zero, right?

There is no interop of Level-Zero with OpenCL. The Level-Zero interop with SYCL is through get_native() methods.

**Review by smaslov-intel — DISMISSED** [review]
LGTM

File: sycl/source/detail/platform_impl.hpp [hunk 1]
```diff
@@ -76,11 +76,10 @@ class platform_impl {
 
   /// \return an instance of OpenCL cl_platform_id.
   cl_platform_id get() const {
-    if (is_host() || getPlugin().getBackend() != cl::sycl::backend::opencl) {
-      throw invalid_object_error(
-          "This instance of platform doesn't support OpenCL interoperability.",
-          PI_INVALID_PLATFORM);
-    }
+    if (is_host())
```
Comment (smaslov-intel) [inline sycl/source/detail/platform_impl.hpp]: why did you remove the BE check here?

Comment (alexanderfle) [reply sycl/source/detail/platform_impl.hpp]: Because platform::get() method is used internally in creating of sycl::context and it comes to exception there.
But now, I found a solution and restore this check.

**Review by smaslov-intel — DISMISSED** [review]
LGTM

**Review by smaslov-intel — DISMISSED** [review]

File: sycl/source/detail/scheduler/commands.cpp [hunk 2]
```diff
@@ -2010,7 +2010,6 @@ cl_int ExecCGCommand::enqueueImp() {
     ExecInterop->MInteropTask->call(InteropHandler);
     Plugin.call<PiApiKind::piEnqueueEventsWait>(MQueue->getHandleRef(), 0,
                                                 nullptr, &Event);
-    Plugin.call<PiApiKind::piQueueRelease>(MQueue->getHandleRef());
```
Comment (smaslov-intel) [inline sycl/source/detail/scheduler/commands.cpp]: Where is the correct instance of the call to "piQueueRelease"? Can you cover this by a test?

Comment (alexanderfle) [reply sycl/source/detail/scheduler/commands.cpp]: Actually, getHandleRef() do only 1 thing -  returns native object, so
`Plugin.call<PiApiKind::piQueueRelease>(MQueue->getHandleRef());`  - is not correct, because it does **an extra piQueueRelease call** and that leads to a crash in the test [interop_task.cpp](https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/handler/interop_task.cpp) 

Before my patch, the line looked like this:
`Plugin.call<PiApiKind::piQueueRelease>(
        reinterpret_cast<pi_queue>(MQueue->get()));`
and, the fact is that get() method does 2 things: retains and returns the native object. Let's pay attention that we called release immediately for the returned object in the same line.

As a result, now, and before my patch, we have the correct reference count for the queue object, But before we did extra instructions in this line. I see no reason to add a new test.

Comment (smaslov-intel) [reply sycl/source/detail/scheduler/commands.cpp]: Makes sense (I didn't notice before that the `.get()` was changed to `.getHandleRef()`.

**Review by smaslov-intel — APPROVED** [review]

**Comment by alexanderfle** [comment]
ping @bader
